gapis/api/vulkan/api/renderpass_framebuffer.api

```diff
@@ -344,3 +344,22 @@ cmd void vkCmdEndRenderPass(
 
   AddCommand(commandBuffer, cmd_vkCmdEndRenderPass, mapPos)
 }
+
+sub void storeImageAttachment(u32 attachmentID) {
+  VK_ATTACHMENT_UNUSED := as!u32(0xFFFFFFFF)
+  if attachmentID != VK_ATTACHMENT_UNUSED {
+    attachment := lastDrawInfo().Framebuffer.ImageAttachments[attachmentID]
+    desc := lastDrawInfo().RenderPass.AttachmentDescriptions[attachmentID]
+    switch desc.storeOp {
+      case VK_ATTACHMENT_STORE_OP_STORE: {
+        if desc.initialLayout != desc.finalLayout {
+          transitionImageLayout(attachment.Image, attachment.SubresourceRange, VK_IMAGE_LAYOUT_UNDEFINED, desc.finalLayout)
```

Comment:
So if the `storeOp` is not `VK_ATTACHMENT_STORE_OP_STORE`, can we skip the layout transition?

Reply:
We should do the layout transition regardless, but only write the image subresource if we are storing the image.

Reply:
storeImageAttachment now transitions the image layout regardless of the storeOp.